ui/src/timeMachine/reducers/index.ts

```diff
@@ -142,6 +135,40 @@ export const initialState = (): TimeMachinesState => ({
   },
 })
 
+const getTableProperties = (view, files) => {
+  const csv = files[0]
+  let pointer = 0,
+    ni
+
+  // cut off the first 3 lines
+  for (ni = 0; ni < 3; ni++) {
+    pointer = csv.indexOf('\r\n', pointer) + 2
+  }
+
+  const existing = (view.properties.fieldOptions || []).reduce((prev, curr) => {
+    prev[curr.internalName] = curr
+    return prev
+  }, {})
+
+  csv
+    .slice(pointer, csv.indexOf('\r\n', pointer))
+    .split(',')
+    .filter(o => !existing.hasOwnProperty(o))
```

Comment:
now that we have a hashmap of the existing column names and a list of ones we got from... where ever.. we can forget about the existing ones and have a default fill for the new ones